File: hotel/latihan_hotel/7_2.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import re

df = pd.read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-02-11/hotels.csv')
df = df[df['is_canceled'] == 0]
df['arrival_date'] = pd.to_datetime(df['arrival_date_year'].astype(str) + '-' + df['arrival_date_month'].astype(str) + '-' + df['arrival_date_day_of_month'].astype(str))
df_reservasi_perhari = df.groupby('arrival_date')['is_canceled'].count().reset_index()
print(df_reservasi_perhari)
df_avg_reservasi_harian = df_reservasi_perhari.groupby(pd.Grouper(key='arrival_date', freq='W'))['is_canceled'].mean().reset_index()
print(df_avg_reservasi_harian)

plt.figure(figsize=(10, 4))
sns.lineplot(data=df_avg_reservasi_harian, x='arrival_date', y='is_canceled')
plt.title("Jumlah Reservasi per Hari")
plt.xlabel("Tanggal Tiba")
plt.ylabel("Jumlah Reservasi")
plt.show()

File: hotel/latihan_hotel/7_2.r
library(dplyr)
library(tidyverse)
library(ggplot2)

df_hotel <- read.csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-02-11/hotels.csv', na.strings = c("", "NA", " ", "N/A", "NULL"))
df <- df_hotel %>% mutate(ID = row_number()) %>% select(ID, everything())
df$arrival_date <- ymd(paste(df$arrival_date_year, df$arrival_date_month, df$arrival_date_day_of_month ,sep='-'))

df <- filter(df, is_canceled == 0)

df_reservasi_perhari_awal <- df %>% group_by(arrival_date) %>% summarise(total = sum(!is.na(is_canceled)))
print(head(df_reservasi_perhari_awal))

df_reservasi_perhari <- df_reservasi_perhari_awal %>% group_by(week = floor_date(arrival_date, "week")) %>% summarise(total = mean(total))
print(head(df_reservasi_perhari))

df_reservasi_perhari <- df_reservasi_perhari_awal %>% group_by(week = floor_date(arrival_date-1, "week")) %>% summarise(total = mean(total))
print(head(df_reservasi_perhari))
print(tail(df_reservasi_perhari))


plot(df_reservasi_perhari_awal$arrival_date, df_reservasi_perhari_awal$total,
        type = "l",
        xlab = "Tanggal Tiba",
        ylab = "Jumlah Reservasi",
        main = "Jumlah Reservasi per Hari")